src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -633,26 +687,33 @@ private void processLogsAndUpload() throws InterruptedException {
                                     .build();
 
                             if (startPosition < file.length()) {
-                                unitOfWork.getLogFileInformationList().add(logFileInformation);
+                                componentLogFileInformation.getLogFileInformationList().add(logFileInformation);
+                                logsProcessor.processLogFiles(
+                                        groupAttempt, componentLogFileInformation, logFileInformation, fileByteChannel);
+                            } else if (startPosition == file.length() && !logFileGroup.isActiveFile(file)) {
```

Comment:
why do we need this else if here?. This is being called after the upload occurs by the `handleCloudWatchAttemptStatus` handler. It seems fishy to be updating this here when the upload has not happened

Reply:
This is for the active file is processed completely in the last run, but rotated before the next run. So we will still get the startPosition to the end, but now this file is rotated and finished. So we need to handle the status. 
`handleCloudWatchAttemptStatus` only handle the file got processed, but the file mentioned above has no contents to be uploaded, so it will not get processed.  
This is the root cause of the duplicate logs.

Reply:
Mmmm if the active log file is process completely (meaning all the logs that are available in it) and it gets rotated on the next run. When finding the files and looping through them it computes the hashes and we would get the start position for the correct file. Maybe I am missing something but this shouldn't be required.

Reply:
We can discuss it in another PR. As @jbutler suggests, we will leave this change to a separate PR.

Reply:
This is new functionality you changed... I don't see how discussing it on another PR changes it. Could you provide an example on a scenario this would be required to be here?